docs/Developers/Debug_Bears.rst

```diff
@@ -0,0 +1,122 @@
+Debug Bears
+===========
+
+This document provides an overview of coala's Debugging Interface.
+The Debugging Interface will help users to debug Bear code and step through
+it using the pdb interface.
+
+After invoking coala's debugger it will step into the ``run()`` method of a
+bear regardless of using ``yield`` or not and will step out as soon as a bear
+finish analysis.
+
+Below is the code given for a simple ``HelloWorldBear`` that prints a debug
+message for each file and yield out a message "A HelloworldBear".
+
+.. code:: python
+
+    import logging
+
+    from coalib.bears.LocalBear import LocalBear
+
+
+    class HelloWorldBear(LocalBear):
+        def run(self, filename, file):
+            logging.debug('Hello World! Checking file {}.'.
+                          format(filename))
+
+            yield self.new_result(message="A HelloworldBear",
+                                  file=filename)
+
+After invoking, debugger will step into ``run()`` method of bear using pdb
+``runcall()`` method i.e.
```

Comment:
Implementation detail ;) I would rather mention that it also exits again as soon as `run()` is left.